### ISSUE TITLE
v4.0.5 edit of blob is not saving in table

### ISSUE THREAD

**Issue by sunnyyadav100** [serge-rider/dbeaver#1861]
Hi,

I have upgraded dbeaver EE from 3.8.5 to 4.0.5. 
In 3.8.5, I can directly edit a blob by pressing CTRL+S twice (One for blob and another for table)
but in 4.0.5 it is not happening. I can edit a blob and save it by CTRL+S but table is recognizing this update (also no * is appearing on table name). On refreshing my changes are getting lost.

**Comment by serge-rider** [comment]
Is this about broken shortcut or blob save fails or ..?

Try to save blob value using toolbar icon (or set auto-save option).
Check Error Log view (or log files: https://github.com/serge-rider/dbeaver/wiki/Log-files) for any related errors.

**Comment by sunnyyadav100** [comment]
Here is output of logs files:

In .log file:
!ENTRY org.jkiss.dbeaver.model 2 0 2017-07-11 10:39:50.282
!MESSAGE Can't delete temp folder 'C:\Users\sunyadav\AppData\Local\Temp\.dbeaver-temp1373830013657058329'

In debug log file:
2017-07-11 10:40:17.916 - Can't detect image type: Unsupported or unrecognized format
2017-07-11 10:40:18.881 - Can't detect image type: Unsupported or unrecognized format
2017-07-11 10:40:19.600 - > Start org.eclipse.compare.core [3.6.0.v20160418-1534]

BLOB is a XML. I was able to edit and save that XML in v3.8.5

I am on Windows 10 - looks like a permission issue (based on .log file)

**Comment by serge-rider** [comment]
> Is this about broken shortcut or blob save fails or ..?
> Try to save blob value using toolbar icon (or set auto-save option).

Is this working?

**Comment by sunnyyadav100** [comment]
No, Context tools ( by Alt+`)  - is in auto commit mode.

**Comment by serge-rider** [comment]
Ok. 
What's your database? 
Table DDL?

**Comment by sunnyyadav100** [comment]
Database is oracle. 

If I save the blob (XML) to file and after modifying table is detecting the change and save is enable. 

When I do inline editing (SHIFT+ENTER) then table is not detecting the change.

**Comment by serge-rider** [comment]
![clob-edit](https://user-images.githubusercontent.com/6398845/28086851-10680798-6689-11e7-88ca-51124a556a55.gif)

**Comment by sunnyyadav100** [comment]
As you have showed on refresh it is prompting me for a save after clicking on save button

![dbe](https://user-images.githubusercontent.com/9139037/28087024-a83c616c-669e-11e7-94ce-5e9ab20a3282.png)

That is also fine with me.

**Comment by serge-rider** [comment]
Does green "Save" button become active after you save BLOB value? Does it become disabled after that?

Also you can open Query Manager view to check what actual queries DBeaver executes (note: enable all query types in QM settings).
In test case above it executes following query
```sql
UPDATE OE.WAREHOUSES x
SET x.WAREHOUSE_SPEC=DATA(XMLType)
WHERE x.ROWID='AAAR6CAAFAAAAF/AAA'
```

**Comment by sunnyyadav100** [comment]
So, when I do editing as you shown then below update query is get executed and save button is enable.

!ENTRY org.jkiss.dbeaver.core 1 92 2017-07-12 11:22:54.136
!MESSAGE UPDATE TCR_HUB.C_REPOS_COMPONENT_INSTANCE x
SET x.CONFIG_DATA=DATA(ByteArrayInputStream)
WHERE x.ROWID='AABiuHAAAAAAn01AAG'
!SUBENTRY 1 org.jkiss.dbeaver.core 1 0 2017-07-12 11:22:54.151
!MESSAGE SUCCESS [1]

But when I do editing by Edit Cell (SHIFT+ENTER) , save button is disable and no update query  in log.

**Comment by serge-rider** [comment]
Problems with standalone lob editor was fixed in 4.1 CE, these fixes will be included in 4.1 EE (should be released in July).

**Comment by sunnyyadav100** [comment]
Thanks Serge-rider . This is very helpful.
